### PR TITLE
Adds pyupgrade

### PR DESCRIPTION
**Pull request by Tpt**
pyupgrade is a tool to migrate Python codebases to use the syntactic sugars introduced in the recent Python versions.

https://github.com/asottile/pyupgrade

The current implementation provides both a formatter and a linter. It uses the configured min Python version if available and fallbacks to Python 3+ if not.

There is also an option to disable migration to new typing syntax ("| None", "list[T]"...). This option is currently applied on this repository.

### CODE REVIEW

File: src/kraken/std/python/tasks/pyupgrade_task.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    format: PyUpgradeTask
+
+
+def pyupgrade(*, name: str = "python.pyupgrade", project: Project | None = None, **kwargs: Any) -> PyUpgradeTasks:
+    project = project or Project.current()
+    settings = python_settings(project)
+    files = list(
+        chain.from_iterable(
+            Path(p).glob("**/*.py")
+            for p in (*kwargs.get("additional_files", ()), settings.source_directory, settings.get_tests_directory())
```
Comment (NiklasRosenstein) [inline src/kraken/std/python/tasks/pyupgrade_task.py:91]:
```suggestion
            for p in (*kwargs.pop("additional_files", ()), settings.source_directory, settings.get_tests_directory())
```

With `.pop()`, you won't need to manually `del kwargs["additional_files"]` later.

Comment (Tpt) [reply src/kraken/std/python/tasks/pyupgrade_task.py]: Thank you! I ignored the existence of this method. Fixed.

**Comment by NiklasRosenstein** [comment]
@Tpt Good to go if you add a changelog entry! :)

**Comment by Tpt** [comment]
@NiklasRosenstein Thank you! Done! I have also rebased the PR.

**Pull request merged**